src/core/api/getSwapLiteQuote.ts

```diff
@@ -0,0 +1,71 @@
+import type { SwapError, SwapUnit } from '../../swap/types';
+import { isSwapError } from '../../swap/utils/isSwapError';
+import type { Token } from '../../token';
+import { formatTokenAmount } from '../utils/formatTokenAmount';
+import { getSwapQuote } from './getSwapQuote';
+import type {
+  APIError,
+  GetSwapQuoteParams,
+  GetSwapQuoteResponse,
+} from './types';
+
+type GetSwapLiteQuoteResponse = {
+  response?: GetSwapQuoteResponse;
+  error?: APIError;
+  formattedFromAmount?: string;
+};
+
+type GetSwapLiteQuoteParams = Omit<GetSwapQuoteParams, 'from'> & {
+  fromSwapUnit: SwapUnit;
+  from?: Token;
+};
+
+/**
+ * Fetches a quote for a swap, but only if the from and to tokens are different.
+ */
+
+export async function getSwapLiteQuote({
```

Comment:
Why not just check if to === from and then return early if that's the case?

Reply:
we're fetching three quotes (to <> ETH, to <> USDC, to <> customToken) so we wouldn't want to return early. we want to gracefully handle case where any of those tokens overlap.

Reply:
you can checkout the implementation of this hook [here](https://github.com/coinbase/onchainkit/pull/1729/files#diff-5fdf01a2dda1a451f5038f451d0d90a5ccbd38f2e31835a0470e631049811cc6R273)